design/anova/adding_anova_to_mmrm_design_document.md

```diff
@@ -0,0 +1,181 @@
+| `...` | Other optional fitted model objects inheriting from class `mmrm`. |
+| `test` | Logical indicating whether likelihood ratio tests should be used to compare the fitted models represented by `object` and the object(s) in `...`. Defaults to `TRUE`. |
+| `refit` | Logical indicating if the models should be refitted before comparing. Defaults to `FALSE`. |
+
+## Function outline
+
+### Plain language
+
+#### Special case: supplying only one model fit
+
+If the user provides only one model fit to `anova.mmrm()`, the function
+will calculate and return the significance of the model terms.
+
+#### Comparing multiple model fits
+
+If the user provides more than one model fit, a `tibble` of results is
+produced. This `tibble` will have a row for each model, and the columns
+will contain identifying information as well as standard diagnostics
+(e.g., AIC, log likelihood, etc.).
+
+If the user supplied `test = TRUE`, the model fits are subject to
+further tests to ensure they are nestable. If the data sets for the fits
+are not the same and the user supplied `refit = TRUE`, the data sets are
+harmonized and the models are refitted using the harmonized data set.
+The pairs of models then undergo likelihood ratio testing, and the
+results are augmented into the `tibble` of results.
+
+### Pseudocode
+
+``` r
+anova.mmrm <- function (object, ..., test = TRUE, refit = FALSE) {
+  
+  # If there are objects supplied to "..."
+  if (...length()) {
+    
+    # Gather all objects into one list
+    fits <- list(object, ...)
+    
+    # Ensure all the objects are mmrm fits
+    validate_mmrm_objects(fits)
+    
+    # Calculate the standard diagnostics no matter what
+    results <-
+      tibble(
+        call = lapply(fits, mmrm::component, "call"),
+        Model = row_number(),
+        df = vapply(fits, get_df, integer(1)), 
+        AIC = vapply(fits, AIC, numeric(1)),
+        BIC = vapply(fits, BIC, numeric(1)),
+        logLik = vapply(fits, logLik, numeric(1))
+      )
+    
+    # If the user has requested the likelihood ratio test...
+    if (test) {
+      
+      # Ensure all fits can be nested
+      ensure_same_formula_for_fixed_effects(fits)
+      ensure_same_time_group_subjectID_vars(fits)
+      ensure_nestable_covariance_structures(fits)
+      
+      # Handle differences in underlying data sets
+      data_sets_are_same <- compare_data_underlying_model_fits(fits)
+      if (!data_sets_are_same) {
+        if (refit) {
+          # Create one harmonized data set from all fits' underlying data
+          # Errors out if not possible
+          harmonized_data <- harmonize_underlying_data(fits)
+          
+          # Refit all models using harmonized data
+          fits <- lapply(fits, refit_mmrm, harmonized_data)
+        } else {
+          stop("Data underlying mmrm fits are not all the same and refit = FALSE")
+        }
+      }
+      
+      
+      results <- results |>
```

Comment:
other side remark: please avoid pipe operations in mmrm package code where possible, and use base functions if possible, we don't have this in our https://github.com/openpharma/mmrm/blob/main/.github/CONTRIBUTING.md but is my personal preference 😄 
is more relevant for the production PRs later, but easier if you start prototyping below like that